providers/amazon/pyproject.toml

```diff
@@ -96,11 +97,12 @@ dependencies = [
     "s3fs>=2023.10.0",
 ]
 "python3-saml" = [
-    "python3-saml>=1.16.0",
+    # Python 3 saml is not compatible with Python 3.13 yet, so we pin it to < 3.13
+    "python3-saml>=1.16.0; python_version < \"3.13\"",
```

Comment:
But with this pinning the dependency is then NOT installed when using Python 3.13? What limitations would we have then in Amazon provider?
Do we need to add notes in README of provider for functional limitations? Or should Amazon provider in general be restricted for 3.13 use until the dependency is 3.13 installable?

Reply:
Also this saml has been problematic already https://github.com/xmlsec/python-xmlsec/issues/344 -> depends on xml-sec which appears to be somewhat abandoned. ... I will gladly get rid of it :). cc: @ferruzzi @vincbeck @o-nikolas - is it still needed for Amazon ?  I believe it is used for one of the authentication modes in AWS - but since it is not going to work for Python 3.13, likely another way of running it is necessary.